src/integrationTest/java/org/opensearch/security/rest/WhoAmITests.java

```diff
@@ -169,4 +181,54 @@ public void testWhoAmIPost() {
         }
 
     }
+
+    @Test
+    public void testAuditLogSimilarityWithTransportLayer() {
+        try (TestRestClient client = cluster.getRestClient(AUDIT_LOG_VERIFIER)) {
+            assertThat(client.get(WHOAMI_PROTECTED_ENDPOINT).getStatusCode(), equalTo(HttpStatus.SC_OK));
+
+            assertThat(client.get("_cat/indices").getStatusCode(), equalTo(HttpStatus.SC_OK));
```

Comment:
I will be implementing end-to-end testing as part of https://github.com/opensearch-project/security/issues/2928
This should suffice the current requirement which is to check for similarity between audit messages from REST and Transport layer.

Reply:
Interpreting 'similarity' to be comparing the number of audit entries is hardly scratching the surface of what the feature does. don't you agree?

I found this example from the search operations tests, how about doing at least this for the most important messages that are being generated where the could be a risk of defect?

https://github.com/opensearch-project/security/blob/bd084c89baff0312a1b11b961e8b850b68e83829/src/integrationTest/java/org/opensearch/security/SearchOperationTest.java#L509-L516

Reply:
Agreed, assertExactly check has already been implemented in this test: https://github.com/opensearch-project/security/blob/f1e0f4a2a2c5bd5e2339af0503ab888928845498/src/integrationTest/java/org/opensearch/security/rest/WhoAmITests.java#L122

I will add that check in this test as well. A broader set of tests will be covered in the end-to-end testing.

Reply:
These tests are not precise, I would expect comparison of an audit log entry of one time vs an entry of another time looking at all the fields.  If there is a better way to confirm the correctness of REST AuthZ vs Transport AuthZ lets shift the validation to the right kind of test, how do you want to proceed?

Reply:
Audit logs from REST will have slightly different entries than transport layer, are you suggesting field-by-field comparison of values for keys that exist in both logs?
Audit log message from REST cannot be directly compared with transport layer because it has a few different fields, see the example logs mentioned below.

Reply:
What do you think of comparing the fields that are the same and those that are different explicitly?

Reply:
done.